crates/core/src/sort.rs

```diff
@@ -0,0 +1,541 @@
+/// use serde_json::json;
+///
+/// let mut items = vec![Item::new("b"), Item::new("a")];
+/// let config = json!({
+///    "sortby": [
+///       { "field": "id", "direction": "asc" }
+///   ]
+/// });
+/// let comparator = ItemComparator::new(config).unwrap();
+/// comparator.sort(&mut items);
+/// assert_eq!(items[0].id, "a");
+/// ```
+#[derive(Debug, Clone)]
+pub struct ItemComparator {
+    sort_fields: Vec<SortField>,
+}
+
+impl ItemComparator {
+    /// Creates a new `ItemComparator` from a JSON configuration.
+    ///
+    /// The configuration should be a JSON object with a `sortby` field, which is
+    /// a list of objects with `field` and `direction` fields.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use stac::sort::ItemComparator;
+    /// use serde_json::json;
+    ///
+    /// let config = json!({
+    ///    "sortby": [
+    ///       { "field": "id", "direction": "asc" }
+    ///   ]
+    /// });
+    /// let comparator = ItemComparator::new(config).unwrap();
+    /// ```
+    pub fn new(config: Value) -> Result<Self, serde_json::Error> {
+        let config: SortConfig = serde_json::from_value(config)?;
+        Ok(Self {
+            sort_fields: config.sortby,
+        })
+    }
+
+    /// Sorts a mutable slice of items.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use stac::{Item, sort::ItemComparator};
+    /// use serde_json::json;
+    ///
+    /// let mut items = vec![Item::new("b"), Item::new("a")];
+    /// let config = json!({
+    ///    "sortby": [
+    ///       { "field": "id", "direction": "asc" }
+    ///   ]
+    /// });
+    /// let comparator = ItemComparator::new(config).unwrap();
+    /// comparator.sort(&mut items);
+    /// assert_eq!(items[0].id, "a");
+    /// ```
+    pub fn sort(&self, items: &mut [Item]) {
+        items.sort_by(|a, b| self.compare(a, b));
+    }
+
+    /// Compares two items.
+    pub fn compare(&self, l: &Item, r: &Item) -> Ordering {
+        for sort_field in &self.sort_fields {
+            let ord = compare_field(l, r, &sort_field.field);
+            if ord != Ordering::Equal {
+                return match sort_field.direction {
+                    Direction::Asc => ord,
+                    Direction::Desc => ord.reverse(),
+                };
+            }
+        }
+        Ordering::Equal
+    }
+}
+
+impl Default for ItemComparator {
+    /// Creates a new `ItemComparator` with the default sort order.
+    ///
+    /// The default sort order is `datetime` descending, followed by `id` ascending.
+    fn default() -> Self {
+        Self {
+            sort_fields: vec![
+                SortField {
+                    field: "datetime".to_string(),
+                    direction: Direction::Desc,
+                },
+                SortField {
+                    field: "id".to_string(),
+                    direction: Direction::Asc,
+                },
+            ],
+        }
+    }
+}
```

Comment:
The existance of this default implementation implies (I think) that we could `impl PartialOrd for Item`?